apps/paymaster-admin/src/app/api/auth/user-data/route.ts

```diff
@@ -0,0 +1,27 @@
+import { cacheTag, cacheLife } from "next/cache";
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+
+import { fetchUserPaymasterData } from "../../../../server/paymaster";
+
+const fetchData = async (
+  walletAddress: string,
+): Promise<ReturnType<typeof fetchUserPaymasterData>> => {
+  "use cache";
+  cacheTag("user-data");
+  cacheLife("seconds");
+  return fetchUserPaymasterData(walletAddress);
+};
+
+export const GET = async (request: NextRequest) => {
+  const walletAddress = request.headers.get("x-authenticated-user");
+  if (!walletAddress) {
+    return NextResponse.json("Unauthorized", { status: 401 });
```

Comment:
While a string is technically valid json, you should construct text responses with `new Response(body, { status: 401 })`.